### PR TITLE
Wrap JAX export variables in TF variable, add test for JAX with TF function endpoint

### PR DESCRIPTION
**Pull request by nkovela1**
This PR makes sure the default variable collections are wrapped as TF variables on the JAX backend, preventing variables from being embedded as constants.

Additionally, I've added a basic test for registering a custom endpoint using jax2tf conversion from the JAX backend.

### CODE REVIEW

**Comment by fchollet** [comment]
Thanks for the PR! Please take a look at the test failure:

```
FAILED keras/export/export_lib_test.py::ExportArchiveTest::test_low_level_model_export - AssertionError: assert False

```

File: keras/export/export_lib.py [hunk 1]
```diff
@@ -142,13 +142,28 @@ def track(self, resource):
...
             # that get saved, because the lists are created in __init__.
-            self._tf_trackable.variables += resource.variables
-            self._tf_trackable.trainable_variables += (
-                resource.trainable_variables
-            )
-            self._tf_trackable.non_trainable_variables += (
-                resource.non_trainable_variables
-            )
+            if backend.backend() == "jax":
+                self._tf_trackable.variables += tf.nest.flatten(
```
Comment (fchollet) [inline keras/export/export_lib.py:146]: Should this be `+=`, or just `=`?

Comment (nkovela1) [reply keras/export/export_lib.py:146]: The trackable.variables are defined as empty lists in the __init__, so `+=` appends them to the list.

**Review by fchollet — APPROVED** [review]
LGTM, thanks!

**Pull request merged**